src/cli/cmd/init/mod.rs

```diff
@@ -0,0 +1,311 @@
+            );
+        }
+
+        if Prompt::bool(
+            "Do you want to add the commonly used Nix formatter (nixpkgs-fmt) to your environment?",
+        )? {
+            dev_shell_packages.push(String::from("nixpkgs-fmt"));
+        }
+
+        // Rust projects
+        if project.maybe_rust() && Prompt::bool("This seems to be a Rust project. Would you like to initialize your flake with built-in Rust dependencies?")? {
+            // Add Rust overlay
+            inputs.insert(
+                String::from("rust-overlay"),
+                String::from("github:oxalica/rust-overlay"),
+            );
+            overlay_refs.push(String::from("rust-overlay.overlays.default"));
+
+            // Add an overlay for inferring a toolchain
+            let rust_toolchain_func = String::from(if project.file_exists("rust-toolchain") {
+                "prev.rust-bin.fromRustupToolchainFile ./rust-toolchain"
+            } else if project.file_exists("rust-toolchain.toml") {
+                "prev.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml"
+            } else {
+                // TODO: make this more granular
+                "prev.rust-bin.latest.default"
+            });
+
+            overlay_attrs.insert(String::from("rustToolchain"), rust_toolchain_func);
+            dev_shell_packages.push(String::from("rustToolchain"));
+
+            // Add cargo-* tools
+            for tool in MultiSelect::new(
+                "You can add any of these Cargo tools to your environment if you wish",
+                CARGO_TOOLS.to_vec(),
+            )
+            .prompt()?
+            .iter()
+            {
+                dev_shell_packages.push(format!("cargo-{tool}"));
+            }
+
+            if Prompt::bool("Do you want to add Rust Analyzer to the environment?")? {
+                dev_shell_packages.push(String::from("rust-analyzer"));
+            }
+        }
+
+        // Go projects
+        if project.maybe_golang() && Prompt::bool("This seems to be a Go project. Would you like to initialize your flake with built-in Go dependencies?")? {
+            if inputs.get("nixpkgs").is_none() && Prompt::bool(
+                "You'll need a Nixpkgs input for Go projects. Would you like to add one?",
+            )? {
+                inputs.insert(
+                    String::from("nixpkgs"),
+                    String::from("https://flakehub.com/f/NixOS/nixpkgs/*.tar.gz"), // TODO: make this more granular
+                );
+            }
+            let go_version =
+            Select::new("Select a version of Go", GO_VERSIONS.to_vec()).prompt()?;
+            dev_shell_packages.push(format!("go_1_{go_version}"));
+
+            if Prompt::bool("Would you like to provide a Go package output?")? {
+                let name = Prompt::string("Enter the name of the output", "default")?;
+                packages.insert(name, String::from("pkgs.buildGoPackage {}"));
```

Comment:
might be good to use the same version of `buildGoPackage` as the one in devshell

```suggestion
                packages.insert(name, format!("pkgs.buildGo1{go_version}Package {}"));
```

Reply:
I'm gonna leave package outputs out of scope for this first iteration